decoding/decode.py

```diff
@@ -0,0 +1,15 @@
+from pyzbar.pyzbar import decode
+from utils import crop_img
+from abc import ABC, abstractmethod
+
+class DecodeInterface(ABC):
+    @abstractmethod
+    def decode_p(self, input_img: str, bounding_boxes):
+        pass
```

Comment:
It's better to just call it Decoder. The fact that it's an interface is obvious enough.

Reply:
why `decode_p`? what `p` means?

Reply:
why `input_img` is string? it should be `ndarray` (check your UML diagram)
also, there's only one `bounding_box` in your diagram